internal/config/config.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package config
+
+import (
+	"github.com/edgexfoundry/edgex-cli/internal/service"
+)
+
+var Configuration HostConfiguration
+
+type HostConfiguration struct {
+	CoreServices        Services
+	DeviceServices      Services
+	ApplicationServices Services
+}
+
+type Services map[string]service.Service
+
+func GetCoreService(name string) service.Service {
+	return Configuration.CoreServices[name]
+}
+
+func GetDeviceService(name string) service.Service {
+	return Configuration.DeviceServices[name]
+}
+
+func GetAppService(name string) service.Service {
+	return Configuration.ApplicationServices[name]
+}
+
+func init() {
+	Configuration.CoreServices = Services{
+		"core-metadata": {
+			Host: "localhost",
+			Port: 59881,
+		},
+		"core-data": {
+			Host: "localhost",
+			Port: 59880,
+		},
+		"core-command": {
+			Host: "localhost",
+			Port: 59882,
+		},
+		"core-scheduler": {
+			Host: "localhost",
+			Port: 59861,
+		},
+		"core-notification": {
+			Host: "localhost",
+			Port: 59860,
+		},
+	}
+
+	Configuration.DeviceServices = Services{
+
+		"device-camera": {
+			Host: "localhost",
+			Port: 59985,
+		},
+
+		"device-mqtt": {
+			Host: "localhost",
+			Port: 59982,
+		},
+
+		"device-rest": {
+			Host: "localhost",
+			Port: 59986,
+		},
+
+		"device-modbus": {
+			Host: "localhost",
+			Port: 59901,
+		},
+	}
+
+	Configuration.ApplicationServices = Services{
+
+		"rules-engine": {
+			Host: "localhost",
+			Port: 59701,
+		},
+		"http-export": {
+			Host: "localhost",
+			Port: 59704,
+		},
+		"mqtt-export": {
+			Host: "localhost",
+			Port: 59703,
+		},
+		"functional-tests": {
+			Host: "localhost",
+			Port: 59705,
+		},
+		"push-to-core": {
+			Host: "localhost",
+			Port: 59702,
+		},
+	}
+}
```

Comment:
If we are supporting these the `version` command needs to be specific to each service since they can be release on a different cadence. For example the core/support service are currently 2.0.0, but `device-coap` and `app-service-configurable` are at 2.0.1 version.

Reply:
IMO we need some way to address app service  and device service generically rather than have the above list. Obviously the port would be a required parameter to the command. thinking out loud... Maybe for the `service-name` we accept `device` and `app` or `device:<port>` and `app:<port>`.

Reply:
I've removed these for now and will focus on supporting the core services. We can add a generic way of supporting these services at a later stage as you suggest.